lib/features/thread/data/network/thread_api.dart

```diff
@@ -226,57 +233,98 @@ class ThreadAPI {
 
     final changesEmailInvocation = jmapRequestBuilder.invocation(changesEmailMethod);
 
-    final getMailboxUpdated = GetEmailMethod(accountId)
-      ..addReferenceIds(processingInvocation.createResultReference(
-          changesEmailInvocation.methodCallId,
-          ReferencePath.updatedPath));
+    GetEmailMethod? getEmailUpdated;
+    GetEmailMethod? getEmailCreated;
+    RequestInvocation? getEmailUpdatedInvocation;
+    RequestInvocation? getEmailCreatedInvocation;
 
     if (propertiesUpdated != null) {
-      getMailboxUpdated.addProperties(propertiesUpdated);
+      getEmailUpdated = GetEmailMethod(accountId)
+        ..addReferenceIds(
+            processingInvocation.createResultReference(
+              changesEmailInvocation.methodCallId,
+              ReferencePath.updatedPath,
+            ),
+          )
+        ..addProperties(propertiesUpdated);
+
+      getEmailUpdatedInvocation = jmapRequestBuilder.invocation(getEmailUpdated);
```

Comment:
write a group test cases for this change

Reply:
done